lib/ui/src/components/preview/preview.js

```diff
@@ -187,14 +187,15 @@ const getTools = memoize(10)(
 );
 
 class Preview extends Component {
-  shouldComponentUpdate({ storyId, viewMode, options }) {
+  shouldComponentUpdate({ storyId, viewMode, options, queryParams }) {
     const { props } = this;
 
     return (
       options.isFullscreen !== props.options.isFullscreen ||
       options.isToolshown !== props.options.isToolshown ||
       viewMode !== props.viewMode ||
-      storyId !== props.storyId
+      storyId !== props.storyId ||
+      queryParams !== props.queryParams
```

Comment:
Does this need to be a deep equals?

Reply:
no, customQueryParams is a new instance every time it's changed by setQueryParams

https://github.com/storybooks/storybook/blob/next/lib/api/src/modules/url.ts#L130-L137

Reply:
Oh nice, this also fixed the problem for `addon-contexts` that eject button won't update the url.